src/IconWithLabel/IconWithLabel.test.tsx

```diff
@@ -56,4 +59,63 @@ describe("IconWithLabel tests", () => {
     expect(textStyle.textOverflow).toBe("ellipsis");
     expect(parentStyle.minWidth).toBe("0");
   });
+  test("renders `IconWithLabel` component with `UserLabel` icon prop. Testing if tooltip is shown on hover of the icon and hides it back on unhover", async () => {
+    render(
+      <IconWithLabel
+        icon={<UserLabel label="James Harper" color="#EC407A" />}
```

Comment:
I don't think we should introduce this dependency on the `UserLabel` furthermor `UserLabel` is itself a wrapper of `IconWithLabel`
```suggestion
        icon={<CarMakerLogo />}
```